op-node/rollup/derive/payload_util.go

```diff
@@ -73,13 +74,17 @@ func PayloadToSystemConfig(rollupCfg *rollup.Config, payload *eth.ExecutionPaylo
 		if err != nil {
 			return eth.SystemConfig{}, fmt.Errorf("failed to parse L1 info deposit tx from L2 block: %w", err)
 		}
+		if info.L1FeeScalar == (eth.Bytes32{}) {
+			// translate Ecotone values back into encoded scalar if needed
+			info.L1FeeScalar[0] = 1
+			binary.BigEndian.PutUint32(info.L1FeeScalar[24:28], info.BlobBaseFeeScalar)
+			binary.BigEndian.PutUint32(info.L1FeeScalar[28:32], info.BaseFeeScalar)
+		}
 		return eth.SystemConfig{
-			BatcherAddr:       info.BatcherAddr,
-			Overhead:          info.L1FeeOverhead,
-			Scalar:            info.L1FeeScalar,
-			GasLimit:          uint64(payload.GasLimit),
-			BaseFeeScalar:     info.BaseFeeScalar,
-			BlobBaseFeeScalar: info.BlobBaseFeeScalar,
+			BatcherAddr: info.BatcherAddr,
+			Overhead:    info.L1FeeOverhead,
+			Scalar:      info.L1FeeScalar,
+			GasLimit:    uint64(payload.GasLimit),
```

Comment:
I feel like we still want the high level human readable API for the 2 additional scalar types?

Reply:
I want to try avoid duplication of the variables in the same struct as much as possible. Adding fields adds complexity to the rollup genesis configuration, and issues when both fields are present.